winsup/cygwin/msys2_path_conv.cc

```diff
@@ -340,8 +340,10 @@ path_type find_path_start_and_type(const char** src, int recurse, const char* en
 
     if (*it == '\0' || it == end) return NONE;
 
-    if (!isalnum(*it) && *it != '/' && *it != '\\' && *it != ':' && *it != '-' && *it != '.') {
-        return find_path_start_and_type(move(src, 1), true, end);
+    while (!isalnum(*it) && *it != '/' && *it != '\\' && *it != ':' && *it != '-' && *it != '.') {
+        recurse = true;
+        it = ++*src;
+        if (it == end || *it == '\0') return NONE;
```

Comment:
I wonder whether we should mark this commit via `git commit --fixup 8e6ef491081215ff6f12fbe36851ed6527c384da` so that the next rebase sees them squashed together?

Reply:
Who knows really u kno..